tests/ballerina-test/src/test/java/org/ballerinalang/test/expressions/lambda/IterableOperationsTests.java

```diff
@@ -263,4 +269,40 @@ public void testInStatement() {
         Assert.assertEquals(returns.length, 1);
         Assert.assertEquals(returns[0].stringValue(), "10");
     }
+
+    @Test
+    public void testSelect() {
+        BValue[] returns = BRunUtil.invoke(basic, "testSelect");
+        Assert.assertNotNull(returns);
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertEquals(returns[0].stringValue(),
+                "{data: [{id:1, salary:100.0}, {id:2, salary:200.0}, {id:3, " + "salary:300.0}]}");
+    }
+
+    @Test
+    public void testSelectCompatibleLambdaInput() {
+        BValue[] returns = BRunUtil.invoke(basic, "testSelectCompatibleLambdaInput");
+        Assert.assertNotNull(returns);
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertEquals(returns[0].stringValue(),
+                "{data: [{id:1, salary:100.0}, {id:2, salary:200.0}, {id:3, " + "salary:300.0}]}");
+    }
+
+    @Test
+    public void testSelectCompatibleLambdaOutput() {
+        BValue[] returns = BRunUtil.invoke(basic, "testSelectCompatibleLambdaOutput");
+        Assert.assertNotNull(returns);
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertEquals(returns[0].stringValue(),
+                "{data: [{id:1, salary:100.0}, {id:2, salary:200.0}, {id:3, " + "salary:300.0}]}");
+    }
+
+    @Test
+    public void testSelectCompatibleLambdaInputOutput() {
+        BValue[] returns = BRunUtil.invoke(basic, "testSelectCompatibleLambdaInputOutput");
+        Assert.assertNotNull(returns);
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertEquals(returns[0].stringValue(),
+                "{data: [{id:1, salary:100.0}, {id:2, salary:200.0}, {id:3, " + "salary:300.0}]}");
+    }
```

Comment:
Shall we move these new tests into org.ballerinalang.test.types.table.TableIterationTest class.

Reply:
Done